### PR TITLE
Script to make a testnet nns-dapp upgrade proposal

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
Upgrade proposals need to be tested, primarily for rendering.

We do already have dummy proposals for testing.  They are all small.

# Changes
- Adds a script that proposes to upgrade nns-dapp on a testnet (or in principle localhost, if it has governance).

# Tests
See a proposal this generated: https://qsgjb-riaaa-aaaaa-aaaga-cai.nnsdapp.dfinity.network/v2/#/proposal/155

Note:  Instructions for getting ic-admin are [on the release page](https://www.notion.so/SOP-Deploy-NNS-dapp-to-mainnet-5347d7438dc04b088c61abd0edb3abd4).

### CODE REVIEW

File: scripts/propose-nns-dapp-upgrade [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  --help)
+    print_help
+    exit 0
+    ;;
+  *) NETWORK="$arg" ;;
+  esac
+done
+
+NETWORK="${1:-testnet}"
+WASM_FILE=nns-dapp.wasm
```
Comment (lmuntanerDfinity) [inline scripts/propose-nns-dapp-upgrade:30]: what happens if the `nns-dapp.wasm` does not exist?

**Review by lmuntanerDfinity — COMMENTED** [review]
A few suggestions and questions

File: scripts/propose-nns-dapp-upgrade (hunk 1 above)
Comment (bitdivine) [reply scripts/propose-nns-dapp-upgrade:30]: The ic-admin call will fail with a message:
```
$ ./scripts/propose-nns-dapp-upgrade 
sha256sum: nns-dapp.wasm: No such file or directory
```
Per the instructions, the user needs to install ic-admin and build the wasm.   I could print the help message if the script fails for any reason.  Done:
```
$ ./scripts/propose-nns-dapp-upgrade
sha256sum: nns-dapp.wasm: No such file or directory


ERROR: exiting with code 1

USAGE: Propose to upgrade a testnet NNS-dapp canister to the current build.
USAGE: 
USAGE: Usage:
USAGE: - Build the nns-dapp canister, e.g. with docker as in the README.md
USAGE: - Put a copy of ic-admin to match the testnet in the root of the repository
USAGE: - Run: ./propose-nns-dapp-upgrade testnet
```

Comment (bitdivine) [reply scripts/propose-nns-dapp-upgrade:30]: Or, yes, as in your other message we can run docker build.  Adding that as well.

Comment (bitdivine) [reply scripts/propose-nns-dapp-upgrade:40]: Doing the same auto-install for ic-admin is not so easy, as we need the commit used to deploy the testnet, ideally, or to scan through the commits of the ic repo to find the most recent blessed release as that is probably close enough.

File: scripts/propose-nns-dapp-upgrade [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+set -euo pipefail
+cd "$(dirname "$(realpath "$0")")/.."
+
+print_help() {
+  cat <<-EOF
+	Propose to upgrade a testnet NNS-dapp canister to the current build.
+
+	Usage:
+	- Build the nns-dapp canister, e.g. with docker as in the README.md
+	- Put a copy of ic-admin to match the testnet in the root of the repository
```
[the remark this answers is not in the document]
Comment (bitdivine) [reply scripts/propose-nns-dapp-upgrade:10]: Added a prompt as well:
```
$ ./scripts/propose-nns-dapp-upgrade
  
  Place a copy of ic-admin in the repo root as per the instructions here:
  https://www.notion.so/SOP-Deploy-NNS-dapp-to-mainnet-5347d7438dc04b088c61abd0edb3abd4
  
Press enter to continue...
```

[the remark this answers is not in the document]
Comment (lmuntanerDfinity) [reply scripts/propose-nns-dapp-upgrade:10]: perfect, thanks!

**Review by lmuntanerDfinity — APPROVED** [review]
LGTM!

**Pull request merged**